parlai/agents/safe_local_human/safe_local_human.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+Agent that gets the local keyboard input in the act() function.
+
+Applies safety classifier(s) to process user and partner messages.
+"""
+
+from parlai.core.message import Message
+from parlai.utils.misc import display_messages
+from parlai.utils.strings import colorize
+from parlai.agents.local_human.local_human import LocalHumanAgent
+from parlai.utils.safety import OffensiveStringMatcher, OffensiveLanguageClassifier
+
+
+class SafeLocalHumanAgent(LocalHumanAgent):
+    @classmethod
+    def add_cmdline_args(cls, argparser):
+        """
+        Add command-line arguments specifically for this agent.
+        """
+        agent = argparser.add_argument_group('Safe Local Human Arguments')
+        agent.add_argument(
+            '--safety',
+            type=str,
+            default='all',
+            choices={'none', 'string_matcher', 'classifier', 'all'},
+            help='Apply safety filtering to messages',
+        )
+        super(SafeLocalHumanAgent, cls).add_cmdline_args(argparser)
+
+    def __init__(self, opt, shared=None):
+        super().__init__(opt)
+        self.id = 'safeLocalHuman'
+        self.init_safety(opt)
+
+    def init_safety(self, opt):
+        if opt['safety'] == 'string_matcher' or opt['safety'] == 'all':
+            self.offensive_string_matcher = OffensiveStringMatcher()
+        if opt['safety'] == 'classifier' or opt['safety'] == 'all':
+            self.offensive_classifier = OffensiveLanguageClassifier()
+        self.self_offensive = False
+
+    def offensive(self, text):
+        if hasattr(
+            self, 'offensive_string_matcher'
+        ) and self.offensive_string_matcher.__contains__(text):
```

Comment:
nit: note that you shouldn't  really call `__contains__` directly, you can actually just write:

```py
text in self.offensive_string_matcher
```